photon-ml/src/test/scala/com/linkedin/photon/ml/optimization/TestObjective.scala

```diff
@@ -14,44 +14,78 @@
  */
 package com.linkedin.photon.ml.optimization
 
-import breeze.linalg.{axpy, Vector, sum}
-import com.linkedin.photon.ml.data.LabeledPoint
-import com.linkedin.photon.ml.function.TwiceDiffFunction
+import breeze.linalg.{Vector, sum}
+import com.linkedin.photon.ml.data.{LabeledPoint, SimpleObjectProvider}
+import com.linkedin.photon.ml.function.{DiffFunction, ObjectiveFunction}
+import com.linkedin.photon.ml.normalization.{NoNormalization, NormalizationContext}
+import com.linkedin.photon.ml.util.Utils
+import org.apache.spark.rdd.RDD
 
 /**
- * Test function used solely to exercise the optimizers.
- * This function has known minimum at {@link TestFunction.CENTROID}.
- * @author bdrew
- */
-
-class TestObjective extends TwiceDiffFunction[LabeledPoint] {
+  * Test objective function used solely to exercise the optimizers.
+  */
+abstract class TestObjective(myData: Either[RDD[LabeledPoint], Iterable[LabeledPoint]])
```

Comment:
@joshvfleming 
This class has been heavily modified for the new objective function class stack, though it's the same in spirit (a toy objective function with an optimal value at known coordinates). It's used by the LBFGS and OWLQN tests now, instead of the ObjectiveFunctionTest.